x-pack/metricbeat/module/panw/system/resources.go

```diff
@@ -0,0 +1,324 @@
+	memoryInfo := parseMemoryInfo(lines[3])
+	swapInfo := parseSwapInfo(lines[4])
+
+	event := mb.Event{
+		Timestamp: timestamp,
+		MetricSetFields: mapstr.M{
+			"uptime": mapstr.M{
+				"days":    systemInfo.Uptime.Days,
+				"hours":   systemInfo.Uptime.Hours,
+				"minutes": systemInfo.Uptime.Minutes,
+			},
+			"user_count": systemInfo.UserCount,
+			"load_average": mapstr.M{
+				"1m":  systemInfo.LoadAverage.OneMinute,
+				"5m":  systemInfo.LoadAverage.FiveMinute,
+				"15m": systemInfo.LoadAverage.FifteenMinute,
+			},
+			"tasks": mapstr.M{
+				"total":    taskInfo.Total,
+				"running":  taskInfo.Running,
+				"sleeping": taskInfo.Sleeping,
+				"stopped":  taskInfo.Stopped,
+				"zombie":   taskInfo.Zombie,
+			},
+			"cpu": mapstr.M{
+				"user":       cpuInfo.User,
+				"system":     cpuInfo.System,
+				"nice":       cpuInfo.Nice,
+				"idle":       cpuInfo.Idle,
+				"wait":       cpuInfo.Wait,
+				"hi":         cpuInfo.Hi,
+				"system_int": cpuInfo.SystemInt,
+				"steal":      cpuInfo.Steal,
+			},
+			"memory": mapstr.M{
+				"total":        memoryInfo.Total,
+				"free":         memoryInfo.Free,
+				"used":         memoryInfo.Used,
+				"buffer_cache": memoryInfo.BufferCache,
+			},
+			"swap": mapstr.M{
+				"total":     swapInfo.Total,
+				"free":      swapInfo.Free,
+				"used":      swapInfo.Used,
+				"available": swapInfo.Available,
+			},
+		},
+		RootFields: mapstr.M{
+			"observer.ip":     m.config.HostIp,
+			"host.ip":         m.config.HostIp,
+			"observer.vendor": "Palo Alto",
+			"observer.type":   "firewall",
+		},
+	}
+
+	events = append(events, event)
+	return events
+}
+
+func parseLoadAverage(line string) SystemLoad {
+	reLoadAvg := regexp.MustCompile(`load average:\s+([\d.]+),\s+([\d.]+),\s+([\d.]+)`)
+	var load1, load5, load15 float64
+
+	if matches := reLoadAvg.FindStringSubmatch(line); matches != nil {
+		load1 = parseFloat("load1", matches[1])
+		load5 = parseFloat("load5", matches[2])
+		load15 = parseFloat("load15", matches[3])
+	}
+
+	return SystemLoad{load1, load5, load15}
+}
+
+func parseUserCount(line string) int {
+	reUserCount := regexp.MustCompile(`(\d+)\s+user`)
+	var userCount int
+	if matches := reUserCount.FindStringSubmatch(line); matches != nil {
+		userCount = parseInt("userCount", matches[1])
+	}
+
+	return userCount
+}
+
+func parseUptime(line string) Uptime {
+	// Uptime less than 1 hour
+	// 	top - 15:03:02 up 4 min,  1 user,  load average: 1.77, 2.74, 1.34
+	// Uptime less than 1 day
+	// 	top - 16:16:29 up  1:18,  1 user,  load average: 0.00, 0.02, 0.01
+	// Uptime 1 day or more
+	// 	top - 11:08:26 up 2 days, 23:02,  1 user,  load average: 0.40, 0.23, 0.37
+
+	// Regular expressions to match different uptime formats
+	// up < 1 hour
+	reMin := regexp.MustCompile(`up\s+(\d+)\s+min`)
+	// up >= 1 hour < 1 day
+	reHourMin := regexp.MustCompile(`up\s+(\d+):(\d+)`)
+	// up >= 1 day
+	reDayHourMin := regexp.MustCompile(`up\s+(\d+)\s+days?,\s+(\d+):(\d+)`)
```

Comment:
This can be done in one regex too make this function more efficient and smaller but let's not do it for now as we have a deadline to meet. Will suggest the improvement in future changes.

Reply:
So do ignore this comment for now.